sqlglot/parser.py

```diff
@@ -1350,8 +1350,24 @@ def _parse_with(self, skip_with_token: bool = False) -> t.Optional[exp.Expressio
 
     def _parse_cte(self) -> exp.Expression:
         alias = self._parse_table_alias()
-        if not alias or not alias.this:
-            self.raise_error("Expected CTE to have alias")
+
+        # Alternative Clickhouse syntax: https://clickhouse.com/docs/en/sql-reference/statements/select/with/
+        if (
+            not alias
+            or not alias.this
+            or not self._match_pair(TokenType.ALIAS, TokenType.L_PAREN, advance=False)
+        ):
+            statement: t.Optional[exp.Expression]
```

Comment:
you don't need this initializer, put it inline 1363, statement: t.Optioanl = self.expression

Reply:
Done